src/main.cc

```diff
@@ -1987,23 +1997,27 @@ watch_child(char *argv[])
                 continue;
 
             if (!mainStartScriptCalled) {
-                mainStartScript(argv[0]);
+                mainStartScript(masterCommand.arg0());
                 mainStartScriptCalled = true;
             }
 
+            // These are only needed by the forked child below, but let's keep
+            // them out of that "no man's land" between fork() and execvp().
+            auto kidCommand = masterCommand;
+            kidCommand.resetArg0(kid.processName().termedBuf());
+            kidCommand.addOption("--kid", kid.gist().termedBuf());
```

Comment:
Is it OK that we simply ignore the existing "--kid" option?
This is the key requirement I was trying to address in my previous implementations. With the ability to duplicate kid options, my previous implementations would be simpler of course. BTW, I think that it would be not easy to simply change your current implementation (based on storing raw strings in argv_) to address this requirement. That is why I was stored parsed array, trying to explain it without luck...

Reply:
You may assert that there is no `--kid` option in masterCommand. We should not reach this function otherwise.

Reply:
Done.